c++/src/kj/async-io-unix.c++

```diff
@@ -1286,7 +1293,8 @@ public:
           // Fine.
           break;
         } else if (error != EINTR) {
-          KJ_FAIL_SYSCALL("connect()", error) { break; }
+          auto address = SocketAddress::toString(addr, addrlen);
```

Comment:
Hmm why the big refactoring of `toString()` into a static method instead of doing:

```
auto address = SocketAddress(addr, addrlen).toString();
```

Reply:
Just a style choice really, avoiding the additional stack allocation. But if the `SocketAddress(addr, addrlen)...` approach is what you prefer then I'm good with that.

Reply:
Well, since it's only called when throwing an exception anyway, I think the copy isn't going to make much difference. I do prefer to minimize code churn when not necessary, so yeah, I think I'd prefer the smaller diff here.

Reply:
Sounds good!